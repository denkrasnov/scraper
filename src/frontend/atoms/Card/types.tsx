import { ReactNode } from "react";

import { DataAttr } from "~app/common/types";

export interface CardProps {
  children: ReactNode;
  dataAttr?: DataAttr;
  href?: string;
  fullHeight?: boolean;
}
