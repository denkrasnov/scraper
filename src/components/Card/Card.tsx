import React, { FC } from "react";

import { CardProps } from "./types";

import styles from "./Card.css";

const Card: FC<CardProps> = props => {
  const { children } = props;

  return <div className={styles.Card}>{children}</div>;
};

export default Card;
