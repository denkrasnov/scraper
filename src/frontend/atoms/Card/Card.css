.root {
  border-radius: 8px;
  box-shadow: 0 1px 5px var(--shadow-color),
    0 4px 6px -2px var(--shadow-color-2);
  background-color: var(--white);
}
