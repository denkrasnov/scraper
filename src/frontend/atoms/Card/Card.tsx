import { FC } from "react";

import mapPropsToStyles from "../helpers/mapPropsToStyles";
import { CardProps } from "./types";
import styles from "./Card.css";

const Card: FC<CardProps> = (props) => {
  const { children, ...restProps } = props;
  const classes = mapPropsToStyles(restProps, styles);

  return <div className={classes}>{children}</div>;
};

export default Card;
