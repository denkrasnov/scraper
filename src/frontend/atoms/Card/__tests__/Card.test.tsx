import React from "react";
import { shallow } from "enzyme";

import Card from "..";

describe("Card", () => {
  it("should render", () => {
    const component = shallow(<Card>Something to display</Card>);

    expect(component).toMatchSnapshot();
  });
});
