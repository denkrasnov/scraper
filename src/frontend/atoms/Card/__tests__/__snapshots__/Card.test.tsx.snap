// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render 1`] = `
<div
  className="Card"
>
  Something to display
</div>
`;
